entity-stats/src/main/java/eu/europeana/entity/stats/service/UsageStatsService.java

```diff
@@ -0,0 +1,151 @@
+            // calculate percentage
+            calculatePercentageValues(entityStatsForLang, entityTotal, entityPerLanguage);
+            entityPerLanguage.setLang(lang);
+            return entityPerLanguage;
+        }
+        return null;
+    }
+
+    private static Query buildSearchQuery(String queryString, String facet) {
+        EntityQueryBuilder queryBuilder = new EntityQueryBuilder();
+        return queryBuilder.buildSearchQuery(queryString, null, queryBuilder.toArray(facet), null, 0, 0,
+                SearchProfiles.facets, null);
+    }
+
+    /**
+     * get the value count from the facets results from Solr
+     *
+     * @param searchQuery
+     * @return
+     */
+    private EntityStats getFacetsResults(Query searchQuery) {
+        List<FacetFieldView> facets = solrEntityService.search(searchQuery, null,null, null ).getFacetFields();
+        if (!facets.isEmpty()) {
+            EntityStats entityStats = new EntityStats();
+            // fetch the first facet result view
+            Map<String, Long> map = facets.get(0).getValueCountMap();
+            for (Map.Entry<String, Long> entry : map.entrySet()) {
+                if(entry.getKey().equals("Agent")) {
+                    entityStats.setAgents(entry.getValue());
+                }
+                if(entry.getKey().equals("Concept")) {
+                    entityStats.setConcepts(entry.getValue());
+                }
+                if(entry.getKey().equals("Place")) {
+                    entityStats.setPlaces(entry.getValue());
+                }
+                if(entry.getKey().equals("Timespan")) {
+                    entityStats.setTimespans(entry.getValue());
+                }
+                if(entry.getKey().equals("Organization")) {
+                    entityStats.setOrganisations(entry.getValue());
+                }
+            }
+            entityStats.setTotal(getTotal(entityStats));
+            return entityStats;
+        }
+        return null;
+    }
+
+    private static void calculatePercentageValues(EntityStats entityPerLanguage, EntityStats entitystatsTotal, EntitiesPerLanguage entities) throws UsageStatsException {
+       entities.setTimespans(getPercentage(entityPerLanguage.getTimespans(), entitystatsTotal.getTimespans()));
+       entities.setPlaces(getPercentage(entityPerLanguage.getPlaces(), entitystatsTotal.getPlaces()));
+       entities.setConcepts(getPercentage(entityPerLanguage.getConcepts(), entitystatsTotal.getConcepts()));
+       entities.setAgents(getPercentage(entityPerLanguage.getAgents(), entitystatsTotal.getAgents()));
+       entities.setOrganisations(getPercentage(entityPerLanguage.getOrganisations(), entitystatsTotal.getOrganisations()));
+       entities.setTotal(getPercentage(entityPerLanguage.getTotal(), entitystatsTotal.getTotal()));
+    }
+
+    private static float getPercentage(float value, float total) throws UsageStatsException {
+      try {
+          if (total != 0.0) {
```

Comment:
better use > 0, the total should be of type long, should be converted to double only for division reasons if needed (I assume that we don't really need the decimals when computing percentages)

Reply:
Total is indeed not in decimals but as we are using the same model for both percentages values (entity per language) and whole values (entity per type), it's set to float now. it was long before but Hugo suggested having decimal values as well. That gives better insights. Though the format is handled in Databox. But I will use > 0 checks